modules/tectonic/resources/tectonic.sh

```diff
@@ -34,120 +24,87 @@ kubectl() {
-kubectl delete -f rbac/role-admin.yaml
-
-kubectl create -f ingress/svc-account.yaml
-kubectl create -f rbac/role-admin.yaml
-kubectl create -f rbac/role-user.yaml
-kubectl create -f rbac/binding-admin.yaml
-kubectl create -f rbac/binding-discovery.yaml
-
-echo "Creating Cluster Config For Tectonic"
-kubectl create -f cluster-config.yaml
-kubectl create -f ingress/cluster-config.yaml
-
-echo "Creating Tectonic Secrets"
-kubectl create -f secrets/pull.json
-kubectl create -f secrets/ingress-tls.yaml
-kubectl create -f secrets/ca-cert.yaml
-kubectl create -f ingress/pull.json
-
-echo "Creating Operators"
-kubectl create -f security/priviledged-scc-tectonic.yaml
-kubectl create -f updater/tectonic-channel-operator-kind.yaml
-kubectl create -f updater/app-version-kind.yaml
-kubectl create -f updater/migration-status-kind.yaml
-
-wait_for_crd tectonic-system channeloperatorconfigs.tco.coreos.com
-kubectl create -f updater/tectonic-channel-operator-config.yaml
-
-kubectl create -f updater/operators/kube-core-operator.yaml
-kubectl create -f updater/operators/tectonic-channel-operator.yaml
-kubectl create -f updater/operators/kube-addon-operator.yaml
-kubectl create -f updater/operators/tectonic-alm-operator.yaml
-kubectl create -f updater/operators/tectonic-utility-operator.yaml
-kubectl create -f updater/operators/tectonic-ingress-controller-operator.yaml
-
-wait_for_crd tectonic-system appversions.tco.coreos.com
-kubectl create -f updater/app_versions/app-version-tectonic-cluster.yaml
-kubectl create -f updater/app_versions/app-version-kube-core.yaml
-kubectl create -f updater/app_versions/app-version-kube-addon.yaml
-kubectl create -f updater/app_versions/app-version-tectonic-alm.yaml
-kubectl create -f updater/app_versions/app-version-tectonic-utility.yaml
-kubectl create -f updater/app_versions/app-version-tectonic-ingress.yaml
-
-# wait for Tectonic pods
+echo "Creating initial roles..."
+kubectl delete --filename rbac/role-admin.yaml
+
+kubectl create --filename ingress/svc-account.yaml
+kubectl create --filename rbac/role-admin.yaml
+kubectl create --filename rbac/role-user.yaml
+kubectl create --filename rbac/binding-admin.yaml
+kubectl create --filename rbac/binding-discovery.yaml
+
+echo "Creating cluster config for Tectonic..."
+kubectl create --filename cluster-config.yaml
+kubectl create --filename ingress/cluster-config.yaml
+
+echo "Creating Tectonic secrets..."
+kubectl create --filename secrets/pull.json
+kubectl create --filename secrets/ingress-tls.yaml
+kubectl create --filename secrets/ca-cert.yaml
+kubectl create --filename ingress/pull.json
+
+echo "Creating operators..."
+kubectl create --filename security/priviledged-scc-tectonic.yaml
+kubectl create --filename updater/tectonic-channel-operator-kind.yaml
+kubectl create --filename updater/app-version-kind.yaml
+kubectl create --filename updater/migration-status-kind.yaml
+
+kubectl --namespace=tectonic-system get customresourcedefinition channeloperatorconfigs.tco.coreos.com
```

Comment:
Now that you've dropped `wait_for_crd`, is the idea that the script will die here if your one-shot request fails, and we'll run through it all again when systemd gives it another pass?

Reply:
This is the equivilent to `wait_for_crd`. `kubelet` will loop until it's successful.